ocean_lib/ocean/ocean_assets.py

```diff
@@ -322,48 +375,88 @@ def create(
-        asset_dict = asset.as_dictionary()
-        ddo_string = json.dumps(asset_dict, separators=(",", ":"))
-        ddo_bytes = ddo_string.encode("utf-8")
-        ddo_hash = create_checksum(ddo_string)
+        document, flags, ddo_hash = self.process_ddo(
+            asset, provider_uri, encrypt_flag, compress_flag
+        )
 
-        # Plain asset
-        if not encrypt_flag and not compress_flag:
-            flags = bytes([0])
-            document = ddo_bytes
+        _ = erc721_token.set_metadata(
+            metadata_state=0,
+            metadata_decryptor_url=provider_uri,
+            metadata_decryptor_address=publisher_wallet.address,
+            flags=flags,
+            data=document,
+            data_hash=ddo_hash,
+            data_proofs=[],
+            from_wallet=publisher_wallet,
+        )
 
-        # Only compression, not encrypted
-        elif compress_flag and not encrypt_flag:
-            flags = bytes([1])
-            # Compress DDO
-            document = lzma.compress(ddo_bytes)
+        # Fetch the asset on chain
+        asset = self._get_aquarius(self._metadata_cache_uri).wait_for_asset(did)
 
-        # Only encryption, not compressed
-        elif encrypt_flag and not compress_flag:
-            flags = bytes([2])
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=ddo_string,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
-            )
-            document = encrypt_response.text
+        return asset
 
-        # Encrypted & compressed
-        else:
-            flags = bytes([3])
-            # Compress DDO
-            compressed_document = lzma.compress(ddo_bytes)
+    def update(
+        self,
+        asset: Asset,
+        publisher_wallet: Wallet,
+        provider_uri: Optional[str] = None,
+        encrypt_flag: Optional[bool] = False,
+        compress_flag: Optional[bool] = False,
+    ) -> str:
+        """Update an asset on-chain.
 
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=compressed_document,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
+        :param asset: The updated asset to update on-chain
+        :param publisher_wallet: Wallet of the publisher updating this asset.
+        :param provider_uri: str URL of service provider. This will be used as base to construct the serviceEndpoint for the `access` (download) service
+        :param encrypt_flag: bool for encryption of the DDO.
+        :param compress_flag: bool for compression of the DDO.
+        :return: update tx_result
+        """
+
+        # Metadata sanity check
+        assert isinstance(
+            asset.metadata, dict
+        ), f"Expected asset.metadata of type dict, got {type(asset.metadata)}"
+
+        asset_type = asset.metadata.get("type")
+
+        assert asset_type in (
```

Comment:
these asserts seem identical to what we have in create. How about a function called _assert_ddo_metadata to keep it DRY?